### PR TITLE
Matt/action handler integration

### PR DESCRIPTION
**Pull request by PaperArmada**
Integrating action handler into the agent execution flow.

### CODE REVIEW

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
super minor edits. remove unused code snippits and functions. 

update code for pylint. 

will run code next. and provide feedback if code doesnt run thx

File: fonpr/agent.py [hunk 1]
```diff
@@ -57,33 +61,86 @@ def print_lim_reqs():
...
+                        prog="FONPR_Agent",
+                        description="Executes policy implementation for closed loop 5G network control.")
+    
+    parser.add_argument(
+            'interval',
+            metavar="I",
+            type=int,
+            default=15,
+            help='Time in minutes between executions of the policy logic.')
+    parser.add_argument(
```
Comment (PaperArmada) [inline fonpr/agent.py:135]: I wanted to build in the ability to update the Prometheus server endpoint at runtime, which this seems to do just fine, but it feels a little clunky to me. If there is a best practice here that I'm missing (or my mind is just running slow today and an obvious solution), please chime in.

**Comment by vinayakSharm** [comment]
Can you please add to the macro read me how to change the file that the agent is updating on github. This should be part of the Action Handler section.

**Comment by vinayakSharm** [comment]
Hi matt here are the steps I took to test your PR:

(1) I cloned your branch. 

(2) I then compiled the docker image locally from your branch

(3) pushed the docker image to dockerhub: teamrespons/respons_agent:matt.pr.test

(4) I then created a new manifest file: https://raw.githubusercontent.com/DISHDevEx/fonpr/vinny/action-handler-integration-pr/deployment/respons_agent_manifest.yml

(5) I deployed the agent to the EKS cluster

Here are my errors at step 5. The agent is deployed in the cluster but the logs say the following: 

'''
usage: FONPR_Agent [-h] I E
FONPR_Agent: error: the following arguments are required: I, E
'''

As a user I do not know how to proceed. Pls update Macro read me, and automate the process of having an agent deployed in the cluster. 

Thanks

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
Hi matt here are the steps I took to test your PR:

(1) I cloned your branch. 

(2) I then compiled the docker image locally from your branch

(3) pushed the docker image to dockerhub: teamrespons/respons_agent:matt.pr.test

(4) I then created a new manifest file: https://raw.githubusercontent.com/DISHDevEx/fonpr/vinny/action-handler-integration-pr/deployment/respons_agent_manifest.yml

(5) I deployed the agent to the EKS cluster

Here are my errors at step 5. The agent is deployed in the cluster but the logs say the following: 

'''
usage: FONPR_Agent [-h] I E
FONPR_Agent: error: the following arguments are required: I, E
'''

As a user I do not know how to proceed. Pls update Macro read me, and automate the process of having an agent deployed in the cluster. 

Thanks

File: fonpr/agent.py [hunk 2]
```diff
@@ -57,33 +61,86 @@ def print_lim_reqs():
...
+    """
+    
+    # Retrieve logs and metrics from the cluster using an advisor
+    if prom_endpoint != '':
+        lim_reqs = collect_lim_reqs(prom_endpoint)
+    else:
+        lim_reqs = collect_lim_reqs()
+    
+    # Process advisor output down to specific value update requests
+    targets = [s for s in lim_reqs.keys() if 'amf' in s]
```
[the remark this answers is not in the document]
Comment (PaperArmada) [reply fonpr/agent.py]: Done

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
Hi matt, great work! 

Need logging from agent thanks

File: fonpr/agent.py [hunk 3]
```diff
@@ -57,26 +86,16 @@ def print_lim_reqs():
...
-        ],
-    )
-
-    logging.info("\t" + df_lim_req.to_string().replace("\n", "\n\t"))
-    print(df_lim_req.head(100))
+    return dict_lim_req
 
-
-if __name__ == "__main__":
+def loop_queries() -> None:
```
[the remark this answers is not in the document]
Comment (PaperArmada) [reply fonpr/agent.py]: Done

File: fonpr/agent.py [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
 import time
 import logging
 from collections import defaultdict
 import pandas as pd
 from advisors import PromClient
 from utilities import prom_cpu_mem_queries
+from action_handler import ActionHandler, get_token
 
 def print_lim_reqs():
+    """
```
[the remark this answers is not in the document]
Comment (PaperArmada) [reply fonpr/agent.py]: Done

File: fonpr/README.md
[the remark this answers is not in the document]
Comment (PaperArmada) [reply fonpr/README.md:1]: I'm inclined to leave it, at least for now. I want some breadcrumbs for people to be able to run it for themselves without having us directly involved.

**Comment by vinayakSharm** [comment]
as discussed minor read me updates, and pr is approved. no code changes needed at this point looks solid. 

I would run pylint and black prior to pushing

**Review by vinayakSharm — APPROVED** [review]

**Pull request merged**